notification/src/jsNotificationService.ts

```diff
@@ -3,67 +3,113 @@ import axios from 'axios'
-        const conditionHolds = this.executor.evaluate(notification.condition, notification.data)
-        console.log('Condition is ' + conditionHolds)
-        if (!conditionHolds) { // no need to trigger notification
-        return Promise.resolve()
-        }
-
-        const message = `Pipeline ${notification.pipelineName}(${notification.pipelineId}) ` +
-        `has new data available. Fetch at ${notification.dataLocation}.`
-
-        switch (notification.type) {
-        case 'WEBHOOK':
-            await this.handleWebhook(notification as Webhook)
-            break
-        case 'FCM':
-            await this.handleFCM(notification as Firebase, message)
-            break
-        case 'SLACK':
-            await this.handleSlack(notification as Slack, message)
-            break
-        default:
-            throw new Error('Notification type not implemented.')
-        }
+  /**
+   * Builds the notification message to be sent,
+   * by composing the contents of the transformation event to readable
+   * message
+   * 
+   * @param event event to extract transformation results from 
+   * @returns message to be sent as notification
+   */
+  buildMessage(event: TransformationEvent): string {
+
+    let message: string                       // message to return
+    const jobError = event.jobResult.error    // Error of transformation (if exists)
+
+    /*======================================================
+    *  Build Message for succesfull transformation/pipline
+    *=======================================================*/
+    if (jobError === undefined) {
+      // Build Stats (Time measures for transformation execution)
+      const jobStats = event.jobResult.stats
+      const start = new Date(jobStats.startTimestamp)
+      const end = new Date(jobStats.endTimestamp)
+
+
+      // Build Success Message
+      message = `Pipeline ${event.pipelineName}(Pipeline ID:${event.pipelineId}) ` +
+        `has new data available. Fetch at ${event.dataLocation}.
+
+        Transformation Details:
+              Start: ${start}
+              End:  ${end}
+              Duration: ${jobStats.durationInMilliSeconds} ms`
+
+    } else {
+      /*====================================================
+      *  Build Message for failed transformation/pipline
+      *====================================================*/
+      message = `Pipeline ${event.pipelineName}(Pipeline ID:${event.pipelineId})Failed.
+
+          Details:
+            Line: ${jobError.lineNumber}
+            Message: ${jobError.message}
+            Stacktrace: ${ jobError.stacktrace}`
     }
 
+    return message
+  }
 
-  private async handleWebhook (webhook: Webhook): Promise<void> {
+  private async handleWebhook (webhook: WebHookConfig, message: string): Promise<void> {
     const callbackObject: WebhookCallback = {
-      location: webhook.dataLocation,
+      location: message,
```

Comment:
The `location` attribute should be renamed now that it contains a message (and not just a location).